baseplate/clients/requests.py

```diff
@@ -101,6 +109,42 @@ def http_adapter_from_config(
     return ValidatingHTTPAdapter(**kwargs)
 
 
+PROM_NAMESPACE = "http_client"
+
+# Latency histogram of HTTP calls made by clients
+# buckets are defined above (from 100µs to ~14.9s)
+LATENCY_SECONDS = Histogram(
+    f"{PROM_NAMESPACE}_latency_seconds",
+    "Latency histogram of HTTP calls made by clients",
+    [
+        "http_method",
```

Comment:
The actual endpoint would have way too high cardinality. The spec says "as defined above" and then above it is the definition for http _servers_, so this might be a mistake in the spec. We could maybe put the calling function in there or something, but that's not very "endpoint" :/

Maybe include this in the spec discussion?

Reply:
The [spec has been updated](https://github.snooguts.net/reddit/baseplate.spec/blob/master/component-apis/prom-metrics.md?plain=1#L89) since the initial comment 6 days ago for this to be reflective of the name of the endpoint being hit rather than the full URL

Reply:
Let's make a new issue out of that, since this is a refactor and that would introduce something new. Also because I have no idea how to do it :) 

Btw the spec says "may be empty" which AFAIK prom treats the same way as not set, so this superficially fulfills the spec :D